### PR TITLE
Update interaction details

### PR DESCRIPTION
**Pull request by ztolley**
interaction

sd

- [x] Remove old formatting code and dead code for interactions
- [x] Switch get interaction details to use new API endpoint
- [x] Create new transformers to view interaction and service delivery details
- [x] Change interaction details to use updated component for key-value table
- [x] Get a screen grab with a service delivery

### CODE REVIEW

File: src/apps/interactions/transformers.js [hunk 1]
```diff
@@ -72,6 +72,99 @@ function transformInteractionToListItem ({
   }
 }
 
+function transformInteractionResponsetoViewRecord ({
```
Comment (yeahfro) [inline src/apps/interactions/transformers.js]: I think we can do more with these functions to make them a bit more DRY

File: src/apps/interactions/views/details.njk [hunk 2]
```diff
@@ -1,6 +1,12 @@
 {% extends "_layouts/two-column.njk" %}
 
 {% block main_grid_right_column %}
-  {{ keyvaluetable(interactionDetails, stripey=true, labels=interactionLabels, keyorder=interactionDisplayOrder) }}
-  <a class="button button-secondary" href="{{interaction.id}}/edit">Edit interaction</a>
+   {% component 'key-value-table', variant='striped', items=interactionViewRecord %}
+  <a class="button button-secondary" href="{{interaction.id}}/edit">
+    {% if interaction.kind == 'service_delivery' %}
```
Comment (yeahfro) [inline src/apps/interactions/views/details.njk]: Could we do something to make this smarter?

File: src/apps/interactions/transformers.js (hunk 1 above)
Comment (ztolley) [reply src/apps/interactions/transformers.js]: I had it as one but then most of the function differed, will go back to one

File: src/apps/interactions/repos.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ const config = require('../../../config')
 const authorisedRequest = require('../../lib/authorised-request')
 
 function getInteraction (token, interactionId) {
```
Comment (yeahfro) [inline src/apps/interactions/repos.js]: Based upon current naming conventions, shall we go for `fetchInteraction`?

Comment (ztolley) [reply src/apps/interactions/repos.js]: Yup

File: test/unit/data/interactions/service-delivery.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{
+  "kind": "service_delivery",
+  "id": "7265dc3c-e89d-45ee-8106-d1e370c1c73d",
+  "company": {
+    "id": "dcdabbc9-1781-e411-8955-e4115bead28a",
+    "name": "Samsung"
+  },
+  "interaction_type": {
```
Comment (reupen) [inline test/unit/data/interactions/service-delivery.json]: Need to remove `interaction_type` here.

File: test/unit/apps/interactions/transformers.test.js [hunk 5]
```diff
@@ -117,7 +122,7 @@ describe('Interaction transformers', () => {
       const expected = {
         interaction_type: '72c226d7-5d95-e211-a939-e4115bead28a',
```
Comment (reupen) [inline test/unit/apps/interactions/transformers.test.js]: Looks like this `interaction_type` needs changing to communication_channel?

Comment (ztolley) [reply test/unit/apps/interactions/transformers.test.js]: Agreed but this is outside the scope of this PR. Changes to the form fields are going to be handled by Lee then we'll do a mop up after that for any remaining interaction_type

File: test/unit/data/interactions/service-delivery.json (hunk 4 above)
Comment (ztolley) [reply test/unit/data/interactions/service-delivery.json]: Will do in another ticket, see below.

File: src/apps/interactions/views/details.njk (hunk 2 above)
Comment (ztolley) [reply src/apps/interactions/views/details.njk]: Such as?

File: src/apps/interactions/transformers.js (hunk 1 above)
Comment (ztolley) [reply src/apps/interactions/transformers.js]: Modified to a single transformer now that is able to handle interactions and service deliveries.

**Pull request merged**